### DISCUSSION TITLE
How do you maintain your dotfiles

### DISCUSSION THREAD

**Discussion by reubenwong97** [NvChad/NvChad#2370] [Q&A] [answered]
Hello!

Was looking to ask the community on how they keep track of the custom directory together with the rest of the dotfiles. I store it as a bare git repo like [so](https://www.atlassian.com/git/tutorials/dotfiles). However, since the custom directory is gitignored, I have not been able to track it like the rest of my dotfiles.

Can anyone share how they keep track of theirs / know of a way to get around this? Thanks!

**Comment by siduck** [comment, marked as answer]
@reubenwong97 either git submodule or symlink it somewhere, like 
```bash
mv ~/.config/nvim/lua/custom ~/dotfiles
ln -s ~/dotfiles/custom ~/.config/lua/custom
```

**Reply by mblarsen** [reply]
In case of sub module I would need to maintain a fork right?
Or not push the submodule setup.

**Reply by benzwt** [reply]
I use stow.
https://venthur.de/2021-12-19-managing-dotfiles-with-stow.html

For me, I create my own git repos for the dir "custom", and use stow to do the "ln" job.
Note that, the Nvchad prevent git to track any files under dir custom, please have a look on the .gitignore.
